addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -194,6 +209,14 @@ odoo.define('website.s_website_form', function (require) {
             // All 'hidden if' fields start with d-none
             this.$target[0].querySelectorAll('.s_website_form_field_hidden_if:not(.d-none)').forEach(el => el.classList.add('d-none'));
 
+            // Reset the initial default values.
+            for (const el of this.initialValues) {
+                el.fieldEl.setAttribute('value', el.initialValue);
+                if (!el.hasValueAttr) {
+                    el.fieldEl.removeAttribute('value');
+                }
```

Comment:
See even here, it leads to strange code: you set a value then potentially remove it. While
```javascript
if (value) {
   // setValue
} else {
   // removeValue
}
```
Is really explicit and better.